src/compiler/scala/tools/nsc/transform/patmat/MatchAnalysis.scala

```diff
@@ -888,7 +893,7 @@ trait MatchAnalysis extends MatchApproximation {
               // if uniqueEqualTo contains more than one symbol of the same domain
               // then we can safely ignore these counter examples since we will eventually encounter
               // both counter examples separately
-              case _ if inSameDomain => None
+              case _ if inSameDomain => Some(WildcardExample)
```

Comment:
There is probably a better way to handle this case. I’m not sure what should ever hit the previous version of this – no tests failed, but I’m sure I’m overlooking something.